flink-core/src/test/java/org/apache/flink/api/common/eventtime/AscendingTimestampsWatermarksTest.java

```diff
@@ -30,7 +30,10 @@
 	@Test
 	public void testWatermarkBeforeRecords() {
 		final TestingWatermarkOutput output = new TestingWatermarkOutput();
-		final AscendingTimestampsWatermarks<Object> watermarks = new AscendingTimestampsWatermarks<>();
+		final WatermarkGenerator<Object> watermarks =
+				WatermarkStrategy
+						.forMonotonousTimestamps()
+						.createWatermarkGenerator(() -> null);
```

Comment:
The `() -> null` lambda here is not correct. If you want to pass a `Context` of the value `null`, then pass null here. 

This currently passes in implementation of the context (as a lambda) that returns a null MetricGroup.
The context is not a functional interface, though. Just an interface that happens to have only one method, yet. As soon as someone adds another method, all tests break, even though they should be oblivious to the change.

Reply:
Yes, that's a very good point!